homeassistant/components/media_extractor/__init__.py

```diff
@@ -102,7 +102,10 @@ def get_stream_selector(self):
         ydl = YoutubeDL({"quiet": True, "logger": _LOGGER})
 
         try:
-            all_media = ydl.extract_info(self.get_media_url(), process=False)
+            if ('http' in self.get_media_url()):
+                all_media = ydl.extract_info(self.get_media_url(), process=False)
+            else:
+                all_media = ydl.extract_info("ytsearch: "+self.get_media_url(), process=False)
```

Comment:
What are we expecting when it hits the else block? Ytdlp supports all kinds of services aside from youtube, so I am wondering why we are now setting ytsearch: here

Reply:
With this code I can either stream a video if I use https:\\youtube... or search a song on youtube if the media_url doesn't contain the http string in it...

Reply:
you can put in media_content_id "Summer Hit 2023" and it will search for it... or you can put the https:\\.... link

Reply:
Yes, but if I then put in a SoundCloud link, that won't work.